### ISSUE TITLE
BUG: compute interaction values for catboost

### ISSUE THREAD

**Issue by MLecardonnel** [shap/shap#3416]
### Issue Description

When computing shap interaction values for a catboost model I get an error:
`AssertionError: The background dataset you provided does not cover all the leaves in the model, so TreeExplainer cannot run with the feature_perturbation="tree_path_dependent" option! Try providing a larger background dataset, no background dataset, or using feature_perturbation="interventional".`

But it would work with the shortcut using the C++ version of Tree SHAP as it is done for xgboost or for computing the shap values for catboost:
`model.get_feature_importance(catboost.Pool(X), fstr_type="ShapInteractionValues")`

### Minimal Reproducible Example

```python
import pandas as pd
import catboost
import shap

titanic_df = pd.read_csv("titanic.csv")
y = titanic_df["Survived"]
X = titanic_df.drop(columns=["Survived"])
model = catboost.CatBoostClassifier().fit(X,y)
explainer = shap.TreeExplainer(model)
explainer.shap_interaction_values(X)
```

### Bug report checklist

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest release](https://github.com/shap/shap/releases) of shap.
- [x] I have confirmed this bug exists on the [master branch](https://github.com/shap/shap/blob/master/CONTRIBUTING.md#installing-from-the-master-branch) of shap.
- [ ] I'd be interested in making a PR to fix this bug

### Installed Versions

Clone of the repo to have fix #3325 after release 0.43.0

**Comment by CloseChoice** [comment]
Potentially connected to #3457

**Comment by CloseChoice** [comment]
Thanks for pointing me towards the C++ implementation of catboost interaction values. I followed your suggestion and implemented the catboost C++ short cut. I also tried a bit to get the titanic example to work, but failed, since your example doesn't show how to get the titanic dataset and I always struggled with some other titanic datasets I could find. Would be great to get your feedback on the PR.
